keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -875,3 +913,118 @@ func downloadServiceFile(componentType types.ComponentType, version semver.Versi
 	}
 	return nil
 }
+
+// NewSystemdDbus creates a new system dbus connection to systemd
+func NewSystemdDbus(ctx context.Context) (*dbus.Conn, error) {
+	return dbus.NewSystemConnectionContext(ctx)
+}
+
+// DisableAndStopSystemdUnit provides a wrapper around removing a systemd system dbus connection, reloading
+// the systemd daemon if reload is true, and disabling and stopping the systemd unit.
+func DisableAndStopSystemdUnit(ctx context.Context, unit string, reload bool) error {
+	d, err := NewSystemdDbus(ctx)
+	if err != nil {
+		return err
+	}
+
+	defer d.Close()
+	err = StopSystemdUnit(ctx, d, unit)
+	if err != nil {
+		return err
+	}
+
+	_, err = d.DisableUnitFilesContext(ctx, []string{unit}, false)
+	if err != nil {
+		return err
+	}
+
+	if reload {
+		err = d.ReloadContext(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+// EnableAndRunSystemdUnit provides a wrapper around creating a new systemd system dbus connection, reloading
+// the systemd daemon if reload is true, and enabling and starting the systemd unit.
+func EnableAndRunSystemdUnit(ctx context.Context, unit string, reload bool) error {
+	d, err := NewSystemdDbus(ctx)
+	if err != nil {
+		return err
+	}
+
+	defer d.Close()
+
+	if reload {
+		err = d.ReloadContext(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	enabled, _, err := d.EnableUnitFilesContext(ctx, []string{unit}, false, true)
+	if err != nil {
+		return err
+	}
+
+	if !enabled {
+		return fmt.Errorf("enable %s failed: %s", unit, err)
+	}
+
+	err = StartSystemdUnit(ctx, d, unit)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+// StartSystemddUnit starts systemd unit (systemctl start unit)
+func StartSystemdUnit(ctx context.Context, d *dbus.Conn, unit string) error {
+	doneChan := make(chan string)
+	defer close(doneChan)
+
+	_, err := d.StartUnitContext(ctx, unit, "replace", doneChan)
+	if err != nil {
+		return err
+	}
+
+	result := <-doneChan
+	if result != "done" {
+		return fmt.Errorf("failed to start %s: %s", unit, result)
+	}
+
+	return nil
+}
+
+// StopSystemddUnit stops systemd unit (systemctl stop unit)
+func StopSystemdUnit(ctx context.Context, d *dbus.Conn, unit string) error {
+	doneChan := make(chan string)
+	defer close(doneChan)
+
+	_, err := d.StopUnitContext(ctx, unit, "replace", doneChan)
+	if err != nil {
+		return err
+	}
+
+	result := <-doneChan
+	if result != "done" {
+		return fmt.Errorf("failed to stop %s: %s", unit, result)
+	}
+
+	return nil
+}
+
+func CheckServiceSystemd(serviceName string) error {
+	status, err := checkServiceStatus("Active", serviceName)
```

Comment:
what's the precise status if one service is running successfully. "Active" or "active" or "running"

Reply:
They break that up into two different fields, https://pkg.go.dev/github.com/coreos/go-systemd@v0.0.0-20191104093116-d3cd4ed1dbcf/dbus#UnitStatus 

`ActiveState` returns either `active` or `inactive` 
`SubState` returns either `running` or `dead`

Here is how you can check that yourself https://go.dev/play/p/8f1dPvhuWq0